### PR TITLE
Support vault-k8s internal leader election

### PR DESCRIPTION
**Pull request by tvoran**
Adds extra RBAC and removes leader-elector container deployment to support https://github.com/hashicorp/vault-k8s/pull/271.

### CODE REVIEW

**Review by tomhjp — APPROVED** [review]

File: test/acceptance/injector-leader-elector.bats [hunk 1]
```diff
@@ -22,21 +22,16 @@ load _helpers
   tries=0
   until [ $tries -ge 60 ]
   do
-    leader="$(echo "$(kubectl exec ${pods[0]} -c sidecar-injector -- wget --quiet --output-document - localhost:4040)" | jq -r .name)"
-    [ -n "${leader}" ] && break
-    ((tries++))
+    owner=$(kubectl get configmaps vault-k8s-leader -o json | jq -r .metadata.ownerReferences\[0\].name)
+    leader=$(kubectl get pods $owner -o json | jq -r .metadata.name)
+    [ -n "${leader}" ] && [ "${leader}" != "null" ] && break
+    let "tries=tries+1"
```
Comment (benashz) [inline test/acceptance/injector-leader-elector.bats:28]: You prefer to use `let` instead of the `++` operator?

**Review by benashz — APPROVED** [review]
Only one question, otherwise 👍

File: test/acceptance/injector-leader-elector.bats (hunk 1 above)
Comment (tvoran) [reply test/acceptance/injector-leader-elector.bats:28]: I prefer whatever works, and this works 😄

Comment (tvoran) [reply test/acceptance/injector-leader-elector.bats:28]: (To explain more, `((tries++))` seems to always throw an error for me under bats.)

**Pull request merged**